src/qililab/platform/platform.py

```diff
@@ -1148,7 +1157,7 @@ def execute_experiment(
 
     def compile_qprogram(
         self, qprogram: QProgram, bus_mapping: dict[str, str] | None = None, calibration: Calibration | None = None
-    ) -> QbloxCompilationOutput | QuantumMachinesCompilationOutput:
+    ) -> tuple(QbloxCompilationOutput | QuantumMachinesCompilationOutput, QdacCompilationOutput | None):  # type: ignore[valid-type]
```

Comment:
Returning `tuple` here will complicate things if we add more instruments e.g. Joel's or other vendors. Why not add add a class e.g. `QProgramCompilationOutput` that contains all instrument-specific compilations as properties? Something like:
```
class QProgramCompilationOutput:
    qblox: QbloxCompilationOutput | None
    qdac: QDacCompilationOutput | None
    quantum_machines: QuantumMachinesCompilationOutput | None
```

So we can easily add more instruments, and the return value of the method is very clear.